streaming/java/streaming-state/src/main/java/org/ray/streaming/state/backend/AbstractStateBackend.java

```diff
@@ -0,0 +1,91 @@
+package org.ray.streaming.state.backend;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.ray.streaming.state.backend.impl.MemoryStateBackend;
+import org.ray.streaming.state.config.ConfigKey;
+import org.ray.streaming.state.keystate.KeyGroup;
+import org.ray.streaming.state.keystate.desc.AbstractStateDescriptor;
+import org.ray.streaming.state.serde.IKMapStoreSerDe;
+import org.ray.streaming.state.store.IKMapStore;
+import org.ray.streaming.state.store.IKVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * AbstractStateBackend. This class is state backend factory.
+ */
+public abstract class AbstractStateBackend implements Serializable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractStateBackend.class);
+
+  protected Map<String, String> config;
+  protected StorageMode storageMode;
+
+  protected BackendType backendType;
+  protected int keyGroupIndex = -1;
+
+  protected AbstractStateBackend(Map<String, String> config) {
+    this.storageMode = StorageMode.getEnum(ConfigKey.getStorageMode(config));
+    this.backendType = BackendType.getEnum(ConfigKey.getBackendType(config));
+    this.config = config;
+  }
+
+  private static AbstractStateBackend getAbstractStateBackend(Map<String, String> config,
+                                                              BackendType type) {
+    switch (type) {
+      case MEMORY:
+        return new MemoryStateBackend(config);
+      default:
+        return new MemoryStateBackend(config);
```

Comment:
At least LOG on mismatch type I think, or throw exception. This can prevent typo issue in configuration.

Reply:
ok, i'll throw a exp here